dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/ConcurrencyTest.java

```diff
@@ -0,0 +1,225 @@
+                   Math.abs(timeElapsed - scheduledDelay) < MARGIN_OF_ERROR);
+    }
+
+    @Test
+    public void testRepeatedTriggerTwentySec() throws Exception {
+        Integer scheduledDelay = 20000;
+
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getVariables().getById("repeatTrigger").setValue("true");
+        config.getVariables().getById("scheduledTime").setValue(scheduledDelay.toString());
+        config.getVariables().getById("repeatManagedExec").setValue("false");
+        server.updateServerConfiguration(config);
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, (server) -> {
+            String result = server.waitForStringInLog("Scheduled thread completed", 20000);
+            assertNotNull("Repeated task not called at checkpoint", result);
+        });
+        server.startServer();
+        Thread.sleep(10000);
+        server.checkpointRestore();
+        String scheduledTaskLog = server.waitForStringInLog("The Liberty server process resumed operation from a checkpoint", 15000);
+        String taskExecutionLog = server.waitForStringInLog("Scheduled thread completed", 20000);
+        long timeElapsed = getTimeElapsed(scheduledTaskLog, taskExecutionLog);
+        assertTrue("Expected and Observed delays differ. Expected: " + scheduledDelay +
+                   " ms | Observed: " + timeElapsed + " ms",
+                   Math.abs(timeElapsed - scheduledDelay) < MARGIN_OF_ERROR);
+    }
+
+    @Test
+    public void testRepeatedMangedExecTenSec() throws Exception {
+        Integer scheduledDelay = 10000;
+
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getVariables().getById("repeatTrigger").setValue("true");
+        config.getVariables().getById("scheduledTime").setValue(scheduledDelay.toString());
+        config.getVariables().getById("repeatManagedExec").setValue("false");
+        server.updateServerConfiguration(config);
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, (server) -> {
+            String result = server.waitForStringInLog("Thread scheduled at startup", 20000);
+            assertNotNull("Repeated task not called at checkpoint", result);
+        });
+        server.startServer();
+        Thread.sleep(10000);
+        server.checkpointRestore();
+        String scheduledTaskLog = server.waitForStringInLog("The Liberty server process resumed operation from a checkpoint", 15000);
+        String taskExecutionLog = server.waitForStringInLog("Scheduled thread completed", 20000);
+        long timeElapsed = getTimeElapsed(scheduledTaskLog, taskExecutionLog);
+        assertTrue("Expected and Observed delays differ. Expected: " + scheduledDelay +
+                   " ms | Observed: " + timeElapsed + " ms",
+                   Math.abs(timeElapsed - scheduledDelay) < MARGIN_OF_ERROR);
+    }
+
+    @Test
+    public void testRepeatedMangedExecTwentySec() throws Exception {
```

Comment:
I've not seeing the point of having both this test and `testRepeatedTriggerTwentySec` they have the same preconditions. The only diff I see is that you are checking for different strings in the pre-restore lambda where both strings are expected. If that was the intent, then this could be combined into one test.